### PR TITLE
[MRG] Separate entity checks from suffix to allow BIDSpath to represent dataset_description.json file

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #727 

I think those checks still should be applied to BIDs entities, but the suffix is allowed to be "overwritten".

It should be noted that a consequence is the ability to represent arbitrary suffixes again if one really is "hacking". E.g. `suffix=desc-derivative_ieeg, check=False` will work, but isn't obviously a documented or intended usage rn.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by agramfort** [comment]
CIs complain @adam2392 

can you see what's going on? thx

**Comment by adam2392** [comment]
https://4228-89170358-gh.circle-artifacts.com/0/dev/generated/mne_bids.BIDSPath.html#mne_bids.BIDSPath

**Review by agramfort — APPROVED** [review]
I'll let @hoechenberger merge if he's happy

File: mne_bids/path.py
Comment (sappelhoff) [inline mne_bids/path.py:595]: Isn't this equivalent?

```suggestion
            if key == 'suffix' and not self.check:
                continue
            if val is not None and key != 'root':
                _check_key_val(key, val)
```

Looks much simpler to me

**Review by sappelhoff — CHANGES_REQUESTED** [review]

File: mne_bids/path.py
Comment (hoechenberger) [inline mne_bids/path.py:276]:
```suggestion
    To represent a filename such as ``dataset_description.json``,
    one can set ``check=False``, and pass ``suffix='dataset_description'``
    and ``extension='.json'``.
```

File: doc/whats_new.rst
Comment (hoechenberger) [inline doc/whats_new.rst:46]:
```suggestion
- Arbitrary file names can now be represented as a `BIDSPath`` by passing the entire name as ``suffix`` and setting ``check=False``, by `Adam Li`_ (:gh:`729`)
```

**Review by hoechenberger — APPROVED** [review]
LGTM once we're green

**Review by sappelhoff — APPROVED** [review]

**Comment by agramfort** [comment]
@adam2392 you have real CI failures apparently

File: mne_bids/path.py
Comment (sappelhoff) [inline mne_bids/path.py:589]:
```suggestion
                pass
                
```

**Comment by adam2392** [comment]
Shall I merge this then? (woot looks like I have squash and merge ability now :p)

**Comment by hoechenberger** [comment]
Thanks, @adam2392!

**Pull request merged**
